internal/sampling/samplingstrategy/adaptive/aggregator.go

```diff
@@ -150,9 +151,42 @@ func (a *aggregator) HandleRootSpan(span *span_model.Span, logger *zap.Logger) {
 	if service == "" || span.OperationName == "" {
 		return
 	}
-	samplerType, samplerParam := span.GetSamplerParams(logger)
+	samplerType, samplerParam := GetSamplerParams(span, logger)
```

Comment:
does not need to be public
```suggestion
	samplerType, samplerParam := getSamplerParams(span, logger)
```
The aggregator already has access to logger, no need to pass it, if you make the function have a receiver

Reply:
you mean by using `aggregator.postAggregator.logger` ?

Reply:
I mean `a.getSamplerParams(span)` (a method, not a function)

Reply:
yes i understand this but i mean where is the logger function to use in this method? 

the only logger in struct `aggregator` is in `a.postAggregator.logger` logger

Reply:
ah, ok. Never mind then. This is a weird design that needs a cleanup externally - there's no reason for the caller of HandleRootSpan to be passing a logger.

Reply:
let's just make the function private

Reply:
okay

Reply:
I booked https://github.com/jaegertracing/jaeger/issues/6584